### PR TITLE
blowfish

### PR DESCRIPTION
**Pull request by enriquesouza**

### CODE REVIEW

**Review by Rockindash — APPROVED** [review]

File: VultisigApp/VultisigApp/Views/Components/BlowfishWarningInformationNote.swift [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            Image(systemName: "exclamationmark.triangle")
+                .foregroundColor(Color.warningYellow)
+        }
+    }
+    
+    var text: some View {
+        VStack(alignment: .leading, spacing: 8) {
+            
+            if blowfishMessages.isEmpty {
+                Text("Transaction Scanned Successfully! You are cleared to proceed with safety.") // Assuming `message` has a `message` property
```
Comment (johnnyluo) [inline VultisigApp/VultisigApp/Views/Components/BlowfishWarningInformationNote.swift:60]: If the response from blowfish is empty , then please don't show anything , this message give user a sense guarantee , which is not good

Comment (enriquesouza) [reply VultisigApp/VultisigApp/Views/Components/BlowfishWarningInformationNote.swift:60]: Shouldn’t we show we scanned the transaction? I can hide it as well.

File: VultisigApp/VultisigApp/View Models/SendCryptoVerifyViewModel.swift [hunk 2]
```diff
@@ -36,6 +36,35 @@ class SendCryptoVerifyViewModel: ObservableObject {
     func amount(for coin: Coin, tx: SendTransaction) -> BigInt {
         return tx.amountInRaw
     }
+    
+    func blowfishEVMTransactionScan(tx: SendTransaction) async throws -> BlowfishResponse? {
```
Comment (johnnyluo) [inline VultisigApp/VultisigApp/View Models/SendCryptoVerifyViewModel.swift:40]: We also need this information on `KeysignMessageConfirmView` , on pair device , it is important to scan the tx

**Review by johnnyluo — CHANGES_REQUESTED** [review]
Make sure we do the scan on pair device as well, and when blowfish returns empty , then don't show any messages , don't give false guarantee

File: VultisigApp/VultisigApp/Views/Components/BlowfishWarningInformationNote.swift (hunk 1 above)
Comment (johnnyluo) [reply VultisigApp/VultisigApp/Views/Components/BlowfishWarningInformationNote.swift:60]: Well , or you can just say "Transaction scanned by Blowfish" , and make sure it is translated

Comment (enriquesouza) [reply VultisigApp/VultisigApp/Views/Components/BlowfishWarningInformationNote.swift:60]: I will do that "Transaction scanned by Blowfish"

Comment (enriquesouza) [reply VultisigApp/VultisigApp/Views/Components/BlowfishWarningInformationNote.swift]: image

@johnnyluo

File: VultisigApp/VultisigApp/View Models/SendCryptoVerifyViewModel.swift (hunk 2 above)
Comment (enriquesouza) [reply VultisigApp/VultisigApp/View Models/SendCryptoVerifyViewModel.swift:40]: Working on it

Comment (enriquesouza) [reply VultisigApp/VultisigApp/View Models/SendCryptoVerifyViewModel.swift:40]: image

**Comment by enriquesouza** [comment]
> Make sure we do the scan on pair device as well, and when blowfish returns empty , then don't show any messages , don't give false guarantee

Showing Logic

Nothing: When blowfishResponse is nil.
Green Success: When blowfishResponse is not nil and has no warnings.
List of Warnings: When blowfishResponse is not nil and has warnings.

**Review by johnnyluo — APPROVED** [review]
I think this is fine for sending , but I don't think this implementation cover swap / bridging  , given we only support swap through THORChain/1inch/LI.FI at the moment , not sure we need it either

**Comment by enriquesouza** [comment]
For swapping it is a different implementation, we will have to check for approvals as well

**Pull request merged**